recipes/boost/all/conanfile.py

```diff
@@ -140,15 +169,20 @@ def _dependencies(self):
             raise ConanException("Cannot find {}".format(dependencies_filepath))
         return yaml.load(open(dependencies_filepath))
 
-    def _iter_modules(self):
-        tree = {k: v[:] for k, v in self._dependencies["dependencies"].items()}
-        while tree:
-            nodeps = set(k for k, v in tree.items() if not v)
-            if not nodeps:
-                raise ConanException("cyclic dependency tree detected")
-            for nodep in nodeps:
-                yield nodep
-            tree = {k: [d for d in v if d not in nodeps] for k, v in tree.items() if k not in nodeps}
+    def _all_dependent_modules(self, name):
+        dependencies = {name}
+        new_dependencies = self._dependencies["dependencies"][name]
+        while True:
+            len_before = len(dependencies)
+            dependencies.update(new_dependencies)
+            len_after = len(dependencies)
+            if len_before == len_after:
+                break
+            next_new_dependencies = set()
+            for new_dependency in new_dependencies:
+                next_new_dependencies.update(set(self._dependencies["dependencies"][new_dependency]))
+            new_dependencies = next_new_dependencies
+        return dependencies
```

Comment:
```suggestion
        dependencies = {name}
        while True:
        	new_dependencies = {}
        	for dependency in dependencies:
        		new_dependencies.update(set(self._dependencies["dependencies"][dependency]))
        	new_dependencies.update(dependencies)
        	if len(new_dependencies) > len(dependencies):
        		dependencies = new_dependencies
        	else:
        		break
        return dependencies
```

I think this is a bit simpler (and I hope still correct :) )

Reply:
On sight, it looks equivalent.